docs/2-deploying/1-automated-previews.md

```diff
@@ -41,42 +41,49 @@ jobs:
+
+This configuration takes advantage of GitLab environments in order to give you better control and visibility into what environments exist and what's deployed to them. On PR creation, both a GitLab and Architect environment will be created. The component specified in the repository will be registered with the Architect Cloud and deployed to the environment. When the PR is either merged or closed, the GitLab environment will be automatically deleted and the component deployed to the environment in the Architect Cloud will be destroyed.
 
 ```yaml
-preview:
-  stage: deploy
+stages:
+  - preview
+
+default:
   image: docker:latest
   services:
     - docker:dind
-  only:
-    - merge_requests
-  variables:
-    ARCHITECT_ACCOUNT: my-account
-    PREVIEW_PLATFORM: architect
-    PREVIEW_TAG: $CI_MERGE_REQUEST_SOURCE_BRANCH_NAME
   before_script:
     - apk add --update npm git
     - apk add yq --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community
-    - npm i -g @architect-io/cli --unsafe-perm
+    - npm install -g @architect-io/cli@0.8 --unsafe-perm
     - architect login -e $ARCHITECT_EMAIL -p $ARCHITECT_PASSWORD
-  script:
-    # Use the yq library to extract the name of the component
-    - export FULL_COMPONENT_NAME=$(yq r ./architect.yml name)
-    # Remove the account name from the component name lifted from the manifest
-    - export COMPONENT_NAME=${FULL_COMPONENT_NAME##*/}
-    # Register the component
-    - architect register . -t $PREVIEW_TAG
-    # Create a preview environment if one doesn't already exist
-    - architect env:create $COMPONENT_NAME-$CI_MERGE_REQUEST_ID -a $ARCHITECT_ACCOUNT --platform $PREVIEW_PLATFORM || echo "Preview environment already exists. Deploying to it now."
-    # (optional) extract the interfaces from the manifest so we can automatically expose them
-    - >
-      export INTERFACES=""
-      for i in $(yq r --printMode p ./src/product-catalog/architect.yml "interfaces.*"); do
-        export INTERFACES="$INTERFACES-i ${i#"interfaces."}:${i#"interfaces."} "
-      done
-    # Deploy the component to the preview environment
-    - architect deploy $FULL_COMPONENT_NAME:$PREVIEW_TAG -e $COMPONENT_NAME-$CI_MERGE_REQUEST_ID -a $ARCHITECT_ACCOUNT --auto_approve $INTERFACES
+
+deploy_preview:
+  stage: preview
+  script: |
+    architect register architect.yml -t $CI_MERGE_REQUEST_ID
+    architect environment:create $CI_MERGE_REQUEST_ID -a $ARCHITECT_ACCOUNT --platform $ARCHITECT_PLATFORM || true
+    architect deploy --auto_approve -a $ARCHITECT_ACCOUNT -e $CI_MERGE_REQUEST_ID $ARCHITECT_COMPONENT_NAME:$CI_MERGE_REQUEST_ID $ARCHITECT_DEPLOY_FLAGS
+  environment:
+    name: architect/preview/$CI_MERGE_REQUEST_ID
+    url: https://app.architect.io/ryan-cahill-a0/environments/$CI_MERGE_REQUEST_ID/
+    on_stop: destroy_preview
+  rules:
+    - if: $CI_MERGE_REQUEST_ID
+
+destroy_preview:
+  stage: preview
```

Comment:
should the stages be the same for preview/destroy_preview?

Reply:
According to the second paragraph under the code block [here](https://docs.gitlab.com/ee/ci/environments/#automatically-stopping-an-environment)